src/notation/internal/notationmidiinput.cpp

```diff
@@ -181,6 +177,10 @@ Note* NotationMidiInput::addNoteToScore(const muse::midi::Event& e)
 
     if (e.opcode() == muse::midi::Event::Opcode::NoteOff) {
         if (isRealtime()) {
+            if (!is.cr()) {
+                return nullptr;
```

Comment:
If we return without doing anything, but still call `undoStack->commitChanges()`, does that create an empty entry on the undo stack? Or does the undo stack have protection against that? (Sorry, I can't check right now)

Reply:
It has the protection, as far as I can tell (if no commands are added, we will rollback)